browser/resources/settings/brave_content_page/containers_icon.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { CrLitElement } from '//resources/lit/v3_0/lit.rollup.js'
+
+import { Icon } from '../containers.mojom-webui.js'
+import { getCss } from './containers_icon.css.js'
+import { getHtml } from './containers_icon.html.js'
+
+export type IconSelectedEvent = CustomEvent<{
+  icon: Icon
+}>
+
+const iconsMap = new Map<Icon, string>([
+  [Icon.kPersonal, 'container-personal'],
+  [Icon.kWork, 'container-work'],
+  [Icon.kShopping, 'container-shopping'],
+  [Icon.kSocial, 'container-social'],
+  [Icon.kEvents, 'container-events'],
+  [Icon.kBanking, 'container-banking'],
+  [Icon.kStar, 'container-star'],
+  [Icon.kTravel, 'container-travel'],
+  [Icon.kSchool, 'container-school'],
+  [Icon.kPrivate, 'container-private'],
+  [Icon.kMessaging, 'container-messaging'],
+]);
+
+export class SettingsBraveContentContainersIconElement extends CrLitElement {
+  static get is() {
+    return 'settings-brave-content-containers-icon'
+  }
+
+  static override get properties() {
+    return {
+      icon: {
+        type: Icon,
```

Comment:
if you change this to `type: Number` does it work?

Reply:
![image](https://github.com/user-attachments/assets/10a31e05-c843-4cef-b1d4-b6c09dfa16e9)

Reply:
Yeah, looks working

```
      icon: {
        type: Number,
        // converter: (value: string | null) => (value ? (+value as Icon) : null),
      },
```

Do you prefer this? I'm not sure if it's better

Reply:
Still, it's really surprising that lit-component convert enum to string :(

Reply:
I think this was happening because `Icon` doesn't have an associated type converter so it defaults to the string converting.

I think `Number` is probably the correct solution here (`type` isn't a typescript type, it's an object which Lit may or may not have a converter for).

Reply:
> type isn't a typescript type, it's an object which Lit may or may not have a converter for).

Aha that makes sense. THanks!

Reply:
> I think Number is probably the correct solution here (type isn't a typescript type, it's an object which Lit may or may not have a converter for).

this is why Lit has a `converter` property to support arbitrary types.

`Number` works, because it's in a list of types that Lit automatically converts from a string. Setting the html attribute value as `icon=${icon}` or `icon="${icon}"` does not make any difference, it's still a string.

For code correctness and for anyone who will read this in future please keep the correct `Icon` type.